EPPs/common.py

```diff
@@ -71,9 +71,12 @@ def open_or_download_file(self, file_or_uid, encoding=None, crlf=False):
             f = open(file_or_uid)
         else:
             a = Artifact(self.lims, id=file_or_uid)
-            f = StringIO(self.get_file_contents(uri=a.files[0].uri, encoding=encoding, crlf=crlf))
-
-        self.open_files.append(f)
+            if a.files:
+                f = StringIO(self.get_file_contents(uri=a.files[0].uri, encoding=encoding, crlf=crlf))
+            else:
+                f = None
+        if f:
+            self.open_files.append(f)
```

Comment:
In step where multiple files can be uploaded the Artifact id exists and is passed to the script regardless of the existence of a file. users of `open_or_download_file` will have to check that the open file returned is not `None`.